File: 1.2_Simple_UI/app/src/main/java/com/example/simpleui/MainActivity.java
package com.example.simpleui;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.RelativeLayout;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    private TextView textView;
    private Button button;
    private int count = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.counter_layout);

        button = findViewById(R.id.my_button);
        textView = findViewById(R.id.my_text);

        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                increase(view);
            }
        });

    }


    public void increase(View view){
        count++;
        textView.setText(String.valueOf(count));
    }
}

File: 9.1_GuessWord/app/src/main/java/com/example/a91_guessword/MainActivity.kt
package com.example.a91_guessword

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import androidx.lifecycle.ViewModelProvider
import kotlinx.android.synthetic.main.activity_main.*


// Our View in MVVM (Model View ViewModel) architecture pattern
class MainActivity : AppCompatActivity() {

    //private var count = 0
    private lateinit var scoreViewModel: ScoreViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        scoreViewModel = ViewModelProvider(this).get(ScoreViewModel::class.java)
        text_view_count.text = scoreViewModel.retrieveScore()

        // You can shorten it using lambda
        button_increase.setOnClickListener(object : View.OnClickListener {
            override fun onClick(p0: View?) {
                //count++
                scoreViewModel.increaseBy(1)
                text_view_count.text = scoreViewModel.retrieveScore()
            }
        })
    }
}